src/charm.py

```diff
@@ -325,5 +353,29 @@ def _push_ca_cert_to_container(self, ca_certificate: str):
             f.write(ca_certificate)
 
 
+def parse_time_string(time_str: str) -> timedelta:
+    """Parse a given time string.
+
+    It must be a number followed by either an
+    s for seconds, h for hours, d for days or y for years.
+
+    Args:
+        time_str: the input string. Ex: "15s", "365d", "10y"
+    Returns:
+        timedelta object representing the given string
+    """
+    value, unit = int(time_str[:-1]), time_str[-1]
+    if unit == "s":
+        return timedelta(seconds=value)
+    elif unit == "h":
+        return timedelta(hours=value)
```

Comment:
I would lead towards having minutes instead of seconds. Having a cert valid for 1 seconds seems quite wild and I'm not even sure this would work, the charm would likely continuously be in maintenance status.

Reply:
I think it makes sense for testing. I don't think having minutes is valuable, since something like 15 minutes is too long for tests and too short for any real usecase.